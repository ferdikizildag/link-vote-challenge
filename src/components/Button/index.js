import React from 'react';
import { PropTypes } from 'prop-types';
import style from './style.module.scss';

const Button = ({ onClick, title, type, disabled }) => {
    return (
        <button className={`${style.button} ${disabled && style.disabled}`} type={type} onClick={onClick} disabled={disabled}>{title}</button>
    );
};

Button.defaultProps = {
    title: 'No Text',
    type: 'button',
    disabled: false
};

Button.propTypes = {
    title: PropTypes.string,
    type: PropTypes.string,
    onClick: PropTypes.func,
    disabled: PropTypes.bool
};

export default Button;
