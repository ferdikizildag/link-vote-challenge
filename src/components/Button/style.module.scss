@import 'utils/variables.scss';

.button{
    width: 100%;
    background-color:$black;
    color: $white;
    border-radius: 50px;
    height: 50px;
    &.disabled{
        background-color: $lightGray;
    }
}