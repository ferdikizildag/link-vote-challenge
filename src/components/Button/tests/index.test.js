import React from 'react';
import Button from 'components/Button';
import renderer from 'react-test-renderer';

describe('Button', () => {

    const onClickMock = jest.fn();

    let buttonProps = {
        onClick: onClickMock,
        title:'',
        type:'',
        disabled: false
    }

    it('renders without crashing', async () => {
        const component= renderer.create(<Button {...buttonProps}/>)
        let tree=component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});