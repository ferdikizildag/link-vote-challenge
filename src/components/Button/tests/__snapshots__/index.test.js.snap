// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without crashing 1`] = `
<button
  className="button false"
  disabled={false}
  onClick={[MockFunction]}
  type=""
>
  
</button>
`;
